File: app/src/main/java/com/rcdvl/marvel/characterslist/CharacterLoadingListItemCreator.java
package com.rcdvl.marvel.characterslist;

import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.RecyclerView.ViewHolder;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.paginate.recycler.LoadingListItemCreator;
import com.rcdvl.marvel.R;

/**
 * Created by renan on 14/06/16.
 */
public class CharacterLoadingListItemCreator implements LoadingListItemCreator {
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.character_loading_list_item, parent, false);
        return new LoadingViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

    }

    class LoadingViewHolder extends RecyclerView.ViewHolder {

        public LoadingViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/kotlin/com/rcdvl/marvel/characterlist/CharacterLoadingListItemCreator.kt
package com.rcdvl.marvel.characterlist

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.paginate.recycler.LoadingListItemCreator
import com.rcdvl.marvel.R

/**
 * Created by renan on 3/17/16.
 */
class CharacterLoadingListItemCreator : LoadingListItemCreator {
    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int) {

    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RecyclerView.ViewHolder? {
        val inflater = LayoutInflater.from(parent?.context);
        val view = inflater.inflate(R.layout.character_loading_list_item, parent, false);
        return VH(view);
    }
}

class VH(view: View) : RecyclerView.ViewHolder(view) {

}
